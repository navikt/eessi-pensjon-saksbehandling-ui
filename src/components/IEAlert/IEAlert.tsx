import NavHighContrast, { themeKeys, HighContrastLink } from 'nav-hoykontrast'
import PT from 'prop-types'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

export const AlertDiv = styled.div`
  padding: 0.5rem 1rem;
  background-color: ${({ theme }: any) => theme[themeKeys.MAIN_INTERACTIVE_COLOR]};
  a {
    color: ${({ theme }: any) => theme[themeKeys.INVERTED_FONT_COLOR]} !important;
  }
  display: flex;
  justify-content: center;
 `

export interface IEAlertProps {
  highContrast: boolean
  onLinkClick?: () => void
}

const IEAlert: React.FC<IEAlertProps> = ({ highContrast, onLinkClick = () => {} }: IEAlertProps) => {
  // Internet Explorer 6-11
  // @ts-ignore
  const isIE = /* @cc_on!@ */false || !!document.documentMode
  // Edge 20+
  const isEdge = !isIE && !!window.StyleMedia
  const { t } = useTranslation()

  if (!isIE && !isEdge) {
    return <div />
  }

  return (
    <NavHighContrast highContrast={highContrast}>
      <AlertDiv>
        <HighContrastLink
          data-test-id='c-iealert__link-id'
          href='https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx'
          onClick={onLinkClick}
          target='_blank'
        >
          {t('buc:alert-IEAlert')}
        </HighContrastLink>
      </AlertDiv>
    </NavHighContrast>
  )
}

IEAlert.propTypes = {
  highContrast: PT.bool.isRequired,
  onLinkClick: PT.func
}

export default IEAlert
