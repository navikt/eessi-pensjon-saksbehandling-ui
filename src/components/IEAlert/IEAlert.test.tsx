import { AlertDiv, IEAlertProps } from 'components/IEAlert/IEAlert'
import { mount, ReactWrapper } from 'enzyme'
import IEAlert from './IEAlert'

describe('components/IEAlert/IEAlert', () => {
  let wrapper: ReactWrapper
  const initialMockProps: IEAlertProps = {
    highContrast: false,
    onLinkClick: jest.fn()
  }

  it('Render: For IE, show alert', () => {
    // mock IE object
    Object.defineProperty(document, 'documentMode', {
      writable: true,
      value: 'something'
    })
    wrapper = mount(<IEAlert {...initialMockProps} />)
    expect(wrapper.exists(AlertDiv)).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
    Object.defineProperty(document, 'documentMode', {
      writable: true,
      value: undefined
    })
  })

  it('Render: For Edge, show alert', () => {
    // mock IE object
    Object.defineProperty(window, 'StyleMedia', {
      writable: true,
      value: 'something'
    } as any)
    wrapper = mount(<IEAlert {...initialMockProps} />)
    expect(wrapper.exists(AlertDiv)).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
    Object.defineProperty(window, 'StyleMedia', {
      writable: true,
      value: undefined
    } as any)
  })

  it('Render: For others, do not show alert', () => {
    // mock IE object
    wrapper = mount(<IEAlert {...initialMockProps} />)
    expect(wrapper.exists(AlertDiv)).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Handling: clicking logo', () => {
    (initialMockProps.onLinkClick as jest.Mock).mockReset()
    Object.defineProperty(document, 'documentMode', {
      writable: true,
      value: 'something'
    } as any)
    wrapper = mount(<IEAlert {...initialMockProps} />)
    wrapper.find('[data-test-id=\'c-iealert__link-id\']').hostNodes().simulate('click')
    expect(initialMockProps.onLinkClick).toHaveBeenCalled()
  })
})
