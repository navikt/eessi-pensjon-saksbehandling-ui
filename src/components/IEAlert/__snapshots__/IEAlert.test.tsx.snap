// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IEAlert/IEAlert Render: For Edge, show alert 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: inherit;
  line-height: inherit;
  color: !important;
}

.c3:focus {
  background-color: !important;
  color: !important;
}

.c3 svg {
  fill: !important;
  stroke: !important;
}

.c3 svg path {
  stroke: !important;
}

.c3 svg:focus {
  fill: !important;
  stroke: !important;
}

ui-fss.g10[id="c2"] {
  content: "c3,";
}

.c1 {
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 a {
  color: !important;
}

ui-fss.g49[id="c0"] {
  content: "c1,";
}

<IEAlert
  highContrast={false}
  onLinkClick={[MockFunction]}
>
  <styled.div>
    <div
      className="c0 c1"
    >
      <Styled(f)
        data-test-id="c-iealert__link-id"
        href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
        onClick={[MockFunction]}
        target="_blank"
      >
        <f
          className="c2 c3"
          data-test-id="c-iealert__link-id"
          href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
          onClick={[MockFunction]}
          target="_blank"
        >
          <a
            className="lenke c2 c3"
            data-test-id="c-iealert__link-id"
            href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
            onClick={[MockFunction]}
            rel="noopener"
            target="_blank"
          >
            buc:alert-IEAlert
          </a>
        </f>
      </Styled(f)>
    </div>
  </styled.div>
</IEAlert>
`;

exports[`components/IEAlert/IEAlert Render: For IE, show alert 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: inherit;
  line-height: inherit;
  color: !important;
}

.c3:focus {
  background-color: !important;
  color: !important;
}

.c3 svg {
  fill: !important;
  stroke: !important;
}

.c3 svg path {
  stroke: !important;
}

.c3 svg:focus {
  fill: !important;
  stroke: !important;
}

ui-fss.g10[id="c2"] {
  content: "c3,";
}

.c1 {
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 a {
  color: !important;
}

ui-fss.g49[id="c0"] {
  content: "c1,";
}

<IEAlert
  highContrast={false}
  onLinkClick={[MockFunction]}
>
  <styled.div>
    <div
      className="c0 c1"
    >
      <Styled(f)
        data-test-id="c-iealert__link-id"
        href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
        onClick={[MockFunction]}
        target="_blank"
      >
        <f
          className="c2 c3"
          data-test-id="c-iealert__link-id"
          href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
          onClick={[MockFunction]}
          target="_blank"
        >
          <a
            className="lenke c2 c3"
            data-test-id="c-iealert__link-id"
            href="https://navno.sharepoint.com/sites/fag-og-ytelser-fagsystemer/SitePages/Ta-i-bruk-PESYS-Chrome!.aspx"
            onClick={[MockFunction]}
            rel="noopener"
            target="_blank"
          >
            buc:alert-IEAlert
          </a>
        </f>
      </Styled(f)>
    </div>
  </styled.div>
</IEAlert>
`;

exports[`components/IEAlert/IEAlert Render: For others, do not show alert 1`] = `
<IEAlert
  highContrast={false}
  onLinkClick={[MockFunction]}
>
  <div />
</IEAlert>
`;
